libs/common/src/middleware/front-end-error-handler.ts

```diff
@@ -0,0 +1,75 @@
+import { HttpStatusCode } from 'axios';
+import { ErrorRequestHandler, Response } from 'express';
+import { isHttpError } from 'http-errors';
+
+/**
+ * Handles the CSRF token error by redirecting the user to the root of the application.
+ * @param res
+ */
+const handleCsrfTokenError = (res: Response) => {
+  console.error("The user's CSRF token is incorrect, redirecting the user to /.");
+  // Redirects always override the status code with 302
+  // So there is no purpose setting the status code explicitly to a 500
+  // We cannot send the status and then redirect as Express closes and sends the response
+  // as soon as send is called
+  res.redirect('/');
+};
+
+/**
+ * Handles an otherwise unhandled error by logging the error and rendering the problem with service page.
+ * @param error
+ * @param res
+ * @param problemWithServiceTemplate
+ */
+const handleUnhandledError = (error: unknown, res: Response, problemWithServiceTemplate: string) => {
+  console.error('An unhandled error occurred:', error);
+  res.status(HttpStatusCode.InternalServerError);
+  res.render(problemWithServiceTemplate);
+};
+
+const getErrorType = (error: unknown) => {
+  if (isHttpError(error) && error.code === 'EBADCSRFTOKEN') {
+    return 'CSRF_TOKEN_ERROR';
+  }
+  return 'UNHANDLED_ERROR';
+};
+
```

Comment:
Can we export and add unit tests for the above functions? `handleCsrfTokenError`, `handleUnhandledError`, `getErrorType`